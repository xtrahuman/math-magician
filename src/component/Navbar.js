import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'calculator',
    },
    {
      id: 3,
      path: '/Quotes',
      text: 'Quotes',
    },
  ];
  return (
    <nav className="navBar">
      <ul>
        {links.map((link) => (
          <li className="li" key={link.id}>
            <NavLink to={link.path} className={(navData) => (navData.isActive ? 'active-link' : '')}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};
export default Navbar;
