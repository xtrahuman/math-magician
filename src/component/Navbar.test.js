import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';

it('renders correctly', () => {
  const nav = renderer
    .create(
        <BrowserRouter>
        <Navbar/>
        </BrowserRouter>,
      )
    .toJSON();
  expect(nav).toMatchSnapshot();
});