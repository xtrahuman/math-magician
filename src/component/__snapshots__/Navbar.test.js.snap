// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navBar"
>
  <ul>
    <li
      className="li"
    >
      <a
        aria-current="page"
        className="active-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="li"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        calculator
      </a>
    </li>
    <li
      className="li"
    >
      <a
        className=""
        href="/Quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
